components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2419,6 +2424,54 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
         return sequenceList;
     }
 
+    /**
+     * This method is used to initiate the web service calls and cluster messages related to stats publishing status
+     *
+     * @param receiverUrl   event receiver url
+     * @param user          username of the event receiver
+     * @param password      password of the event receiver
+     * @param updatedStatus status of the stat publishing state
+     */
+    public void callStatupdateService(String receiverUrl, String user, String password, boolean updatedStatus) {
+
+        try {
+            gatewyEnvironments = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
+                    getAPIManagerConfiguration().getApiGatewayEnvironments();
+
+            for (String gatewayEnvironmentName : gatewyEnvironments.keySet()) {
+                Environment currentGatewayEnvironment = gatewyEnvironments.get(gatewayEnvironmentName);
+                String gatewayServiceUrl = currentGatewayEnvironment.getServerURL();
+                String gatewayUserName = currentGatewayEnvironment.getUserName();
+                String gatewayPassword = currentGatewayEnvironment.getPassword();
+
+                //get the stub and the call the admin service with the credentials
+                GatewayStatsUpdateServiceStub stub =
+                        new GatewayStatsUpdateServiceStub(gatewayServiceUrl + "GatewayStatsUpdateService");
```

Comment:
Done in : https://github.com/chamindias/carbon-apimgt/commit/2917bb535a4686418310b787af361524ff352252